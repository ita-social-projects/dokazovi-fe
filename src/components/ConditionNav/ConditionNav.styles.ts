import { makeStyles, Theme } from '@material-ui/core';

export const useStyles = makeStyles(
  (theme: Theme) => ({
    root: {
      display: 'flex',
      marginBottom: '80px',
      listStyle: 'none',
      '& :not(:last-child)': {
        marginRight: '20px',
      },
    },
    link: {
      color: '#2F80ED',
      fontFamily: 'Raleway',
      fontWeight: 500,
      lineHeight: '16px',
      textDecoration: 'underline',
    },
    linkSelected: {
      color: '#000000',
      fontWeight: 700,
      textDecoration: 'none',
    },
  }),
  { name: 'ConditionsNav' },
);
