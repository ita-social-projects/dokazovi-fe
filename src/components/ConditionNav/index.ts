import ConditionNav from './ConditionNav';

export default ConditionNav;
