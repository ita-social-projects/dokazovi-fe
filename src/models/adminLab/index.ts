/* eslint-disable import/no-cycle */

export {
  adminLabReducer,
  setStateToInit,
  setSort,
  setFilter,
  setPage,
  setFiltersToInit,
  setField,
} from './reducers';
export { getMaterialsAction, archiveAdminPost } from './asyncActions';
export { selectAdminLab, selectMeta } from './selectors';
