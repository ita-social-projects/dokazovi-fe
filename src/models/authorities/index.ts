/* eslint-disable */
export { getAuthoritiesAsyncAction } from './asyncAction';
export { authoritiesReducer } from './reducers';
export { selectAuthorities } from './selectors';
