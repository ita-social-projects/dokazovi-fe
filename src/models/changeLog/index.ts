/* eslint-disable import/no-cycle */
export { changeLogReducer, setChangesSize } from './reducer';
export { selectSize } from './selectors';
