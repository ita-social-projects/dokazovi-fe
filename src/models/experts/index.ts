/* eslint-disable import/no-cycle */
export {
  fetchExperts,
  fetchExpertById,
  fetchExpertMaterials,
} from './asyncActions';
export { resetMaterials, loadPosts, expertsReducer } from './reducers';
export {
  selectExperts,
  selectLoadingExpertsPosts,
  selectLoadingExperts,
  selectExpertsData,
} from './selectors';
