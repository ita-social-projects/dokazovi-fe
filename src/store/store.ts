import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import rootReducer from './rootReducer';
import type { RootStateType } from './rootReducer';

export const store = configureStore({
  reducer: rootReducer,
});

export type AppThunkType = ThunkAction<
  void,
  RootStateType,
  unknown,
  Action<string>
>;

export type AppDispatchType = typeof store.dispatch;
