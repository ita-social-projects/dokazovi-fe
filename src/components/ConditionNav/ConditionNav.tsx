import React from 'react';
import { useSelector } from 'react-redux';
import { NavLink, useLocation } from 'react-router-dom';
import i18n, { langTokens } from '../../locales/localizationInit';
import { selectHeaderVisibility } from '../../models/headerVisibility';
import { useStyles } from './ConditionNav.styles';

export default function ConditionNav(): JSX.Element {
  const classes = useStyles();
  const location = useLocation();
  const isHeaderVisible = useSelector(selectHeaderVisibility);

  const isActive = (hash: string) => location.hash === hash;

  return (
    <ul
      className={`
      ${classes.root} 
      ${String(isHeaderVisible.visibility ? classes.moveRoot : '')}
    `}
    >
      <li key="about">
        <NavLink
          to="#about"
          className={classes.link}
          data-testid="about"
          activeClassName={isActive('#about') ? classes.linkSelected : ''}
        >
          {i18n.t(langTokens.footer.aboutPlatform)}
        </NavLink>
      </li>
      <li key="rules">
        <NavLink
          to="#rules"
          className={classes.link}
          data-testid="rules"
          activeClassName={isActive('#rules') ? classes.linkSelected : ''}
        >
          {i18n.t(langTokens.footer.termsOfUse)}
        </NavLink>
      </li>
      <li key="contacts">
        <NavLink
          to="#contacts"
          className={classes.link}
          data-testid="contacts"
          activeClassName={isActive('#contacts') ? classes.linkSelected : ''}
        >
          {i18n.t(langTokens.footer.contacts)}
        </NavLink>
      </li>
    </ul>
  );
}
