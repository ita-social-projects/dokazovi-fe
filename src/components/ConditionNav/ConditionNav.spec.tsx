import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { store } from 'models/store';
import ConditionNav from './ConditionNav';

const renderComponent = () =>
  render(
    <Provider store={store}>
      <MemoryRouter>
        <ConditionNav />
      </MemoryRouter>
    </Provider>,
  );

const renderComponentWithHistory = () => {
  const history = createMemoryHistory();

  render(
    <Provider store={store}>
      <Router history={history}>
        <ConditionNav />
      </Router>
    </Provider>,
  );

  return { history };
};

const hasClassContaining = (classes: string[], str: string) => {
  const classContainingStr = classes.find((el) => el.includes(str));
  return !!classContainingStr;
};

describe('ConditionsNav', () => {
  beforeEach(() => renderComponent());

  it('should render correctly and match the snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <ConditionNav />
        </MemoryRouter>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should has "about" link', () => {
    expect(screen.getByText('Про платформу')).toBeInTheDocument();
  });

  it('should has "rules" link', () => {
    expect(screen.getByText('Правила використання')).toBeInTheDocument();
  });

  it('should has "contacts" link', () => {
    expect(screen.getByText('Контакти')).toBeInTheDocument();
  });
});

describe('ConditionsNav Events', () => {
  it('should change query #fragment by click on a link', () => {
    const { history } = renderComponentWithHistory();

    const aboutLink = screen.getByTestId('about');
    const rulesLink = screen.getByTestId('rules');
    const contactsLink = screen.getByTestId('contacts');

    userEvent.click(aboutLink);
    expect(history.location.hash).toBe('#about');

    userEvent.click(rulesLink);
    expect(history.location.hash).toBe('#rules');

    userEvent.click(contactsLink);
    expect(history.location.hash).toBe('#contacts');
  });

  it('should toggle "linkSelected" class on an active link', () => {
    renderComponent();

    const aboutLink = screen.getByTestId('about');
    const rulesLink = screen.getByTestId('rules');
    const contactsLink = screen.getByTestId('contacts');

    userEvent.click(aboutLink);
    expect(hasClassContaining(Array.from(aboutLink.classList), 'linkSelected'));
    expect(
      hasClassContaining(Array.from(rulesLink.classList), 'linkSelected'),
    ).toBeFalsy();
    expect(
      hasClassContaining(Array.from(contactsLink.classList), 'linkSelected'),
    ).toBeFalsy();

    userEvent.click(rulesLink);
    expect(hasClassContaining(Array.from(rulesLink.classList), 'linkSelected'));
    expect(
      hasClassContaining(Array.from(aboutLink.classList), 'linkSelected'),
    ).toBeFalsy();
    expect(
      hasClassContaining(Array.from(contactsLink.classList), 'linkSelected'),
    ).toBeFalsy();

    userEvent.click(contactsLink);
    expect(
      hasClassContaining(Array.from(contactsLink.classList), 'linkSelected'),
    );
    expect(
      hasClassContaining(Array.from(aboutLink.classList), 'linkSelected'),
    ).toBeFalsy();
    expect(
      hasClassContaining(Array.from(rulesLink.classList), 'linkSelected'),
    ).toBeFalsy();
  });
});
