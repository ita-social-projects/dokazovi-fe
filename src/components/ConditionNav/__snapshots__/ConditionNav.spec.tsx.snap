// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionsNav should render correctly and match the snapshot 1`] = `
<DocumentFragment>
  <ul
    class="
      ConditionsNav-root-1 
      
    "
  >
    <li>
      <a
        aria-current="page"
        class="ConditionsNav-link-3"
        data-testid="about"
        href="/#about"
      >
        Про платформу
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="ConditionsNav-link-3"
        data-testid="rules"
        href="/#rules"
      >
        Правила використання
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        class="ConditionsNav-link-3"
        data-testid="contacts"
        href="/#contacts"
      >
        Контакти
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
